### ISSUE TITLE
Should this language server start in PHP files?

### ISSUE THREAD

**Issue by rwols** [sublimelsp/LSP-html#18]
Currently this language server starts in PHP files too. But it has strange trigger characters:

```
{'characters': '.:<"=/', 'selector': 'meta.tag, source - comment - string.quoted.double.block - string.quoted.single.block - string.unquoted.heredoc', 'server': 'LSP-html'}
```

So every time I type `=` in PHP, I get an AC popping up which is undesirable.

This could be fixed either by adding an `auto_complete_selector` for this config, or by not starting vscode-html-language-server in php files at all.

**Comment by rchl** [comment]
php file is technically an HTML code with `<?php ... ?>` tags so it should make sense to start an HTML server on it if it only handles HTML parts and doesn't interject otherwise. But I have no idea how well or not well it actually works in PHP files.

Maybe @jfcherng has more experience

**Comment by rwols** [comment]
The auto_complete_selector should include at least ` - source.php` so that it doesn't trigger inside of `<?php` blocks.

**Comment by rchl** [comment]
The problem with that (or with the handling of that option in LSP) is that once you add the `auto_complete_selector` option, you override the sane defaults. So then you have to duplicate those in your setting.

The options:
 - append user selector into the default - bad, wouldn't always know where to put it I guess
 - have more configurability so that you can append or replace
 - have like a variable placeholder (`$default_selector`) but that also can have issues with ordering things properly maybe

**Comment by jfcherng** [comment]
> php file is technically an HTML code with `<?php ... ?>` tags so it should make sense to start an HTML server on it if it only handles HTML parts and doesn't interject otherwise. But I have no idea how well or not well it actually works in PHP files.
> 
> Maybe @jfcherng has more experience

Modern PHP frameworks are usually in a MVC(-like) structure so PHPers don't write PHP mixed with HTML. Or there is no "View" layer at all. I.e., purely, an API server.

But there are some situations that I can figure out that may mix PHP codes with HTML:

- people who just want/like to mix them for whatever reasons
- WordPress themes

  I saw this from the `intelephense`'s issue tracker for a few times.
  They may also prefer using `<?php if(...): ?>` and `<?php endif; ?>`.

**Comment by rchl** [comment]
Then lets just disable it.

**Comment by rwols** [comment]
Fixed by #19
